api/src/controllers/cle/young.js

```diff
@@ -33,9 +33,26 @@ router.get("/by-classe/:id", passport.authenticate("referent", { session: false,
       }
     }
 
-    const students = await YoungModel.find({ classeId: value })?.lean();
+    const getYoungCountByStatus = async (value) => {
+      const students = await YoungModel.find({ classeId: value })?.lean();
+
+      const statusCount = students.reduce((acc, student) => {
+        const status = student.status;
+        acc[status] = (acc[status] || 0) + 1;
+        return acc;
+      }, {});
+
+      const result = { total: students.length };
+      Object.keys(statusCount).forEach((status) => {
+        result[YOUNG_STATUS[status]] = statusCount[status];
+      });
+
+      return result;
+    };
+
+    const result = await getYoungCountByStatus(value);
```

Comment:
pourquoi rajouter une fonction et pas juste :
```suggestion
      const students = await YoungModel.find({ classeId: value })?.lean();

      const statusCount = students.reduce((acc, student) => {
        const status = student.status;
        acc[status] = (acc[status] || 0) + 1;
        return acc;
      }, {});

      const result = { total: students.length };
      Object.keys(statusCount).forEach((status) => {
        result[YOUNG_STATUS[status]] = statusCount[status];
      });
```